doc/compiler.md

```diff
@@ -0,0 +1,40 @@
+## Miniscript Compiler
+
+This library provides a Policy compiler that converts bitcoin Policy to Miniscript which can
+be used with "compiler" feature. The compiler offers a simple way to interface with
+miniscript and often the first place beginners start to play with miniscript. There are
+however several limitations and caveats to the compiler. This document tries the compiler
+inner working as well as explains the guarantees it offers(or does not offer).
+
+### The Miniscript compiler
+
+As mentioned, the policy compiler takes in input a policy and converts it to a miniscript. The
```

Comment:
...except that "a miniscript" is fine to be uncapitalized because then it's an ordinary noun (like "script") rather than a proper noun.